File: indices-of-substring.py
def indices_of_substring(text):
    target = input()
    result = []
    for i in range(len(text)):
        if text[i:].startswith(target):
            while True:
                index = text[i:i+len(target)].find(target)
                if index == -1:
                    break
                result.append(i + index)
                i += index + len(target)
    return result

File: indices-of-substring.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> indicesOfSubstring(string text, string target) {
    vector<int> result;
    int n = text.length();
    int m = target.length();

    for (int i = 0; i <= n - m; ++i) {
        if (text.substr(i, m) == target) {
            result.push_back(i);
        }
    }

    return result;
}

int main() {
    // Read input from user
    string text;
    int numTargets;
    cin >> text >> numTargets;

    vector<string> targets;
    for(int i = 0; i < numTargets; ++i) {
        string target;
        cin >> target;
        targets.push_back(target);
    }

    vector<int> indices;
    for(string target : targets) {
        indices = indicesOfSubstring(text, target);
        for(int index : indices) {
            cout << index << endl;
        }
    }

    return 0;
}